cypress/integration/vesselCommunications.spec.ts

```diff
@@ -1,57 +1,231 @@
-import { givenIHaveSelected, whenIClickContinue } from "./common.spec";
-
-describe("As a beacon owner, I want to register my communication details so SAR can contact me in an emergency", () => {
+import {
+  andIClickContinue,
+  finallyIClear,
+  finallyIUncheck,
+  givenIAmAt,
+  givenIHaveSelected,
+  iCanClickTheBackLinkToGoToPreviousPage,
+  thenIShouldSeeAnErrorMessageThatContains,
+  thenIShouldSeeAnErrorSummaryLinkThatContains,
+  thenMyFocusMovesTo,
+  thenTheUrlShouldContain,
+  whenIClickContinue,
+  whenIClickOnTheErrorSummaryLinkContainingText,
+  whenIType,
+} from "./common.spec";
+
+describe("As a beacon owner and maritime pleasure vessel user", () => {
   const pageUrl = "/register-a-beacon/vessel-communications";
 
   beforeEach(() => {
-    givenIAmOnTheVesselCommunicationsPage();
+    givenIAmAt(pageUrl);
   });
 
-  it("requires an MMSI number if the fixed VHF checkbox is selected", () => {
-    givenIHaveSelected("#fixedVhfRadio");
-    andIHaveLeftTheRelevantTextInputBlank();
-
-    whenIClickContinue();
+  describe("the Fixed VHF/DSC radio option", () => {
+    it("requires an MMSI number if the fixed VHF/DSC radio checkbox is selected", () => {
+      const expectedErrorMessage = ["We need", "fixed VHF"];
+      givenIHaveSelected("#fixedVhfRadio");
+
+      whenIType(" ", "fixedVhfRadioInput");
+      whenIClickContinue();
+      thenIShouldSeeAnErrorSummaryLinkThatContains(...expectedErrorMessage);
+      thenIShouldSeeAnErrorMessageThatContains(...expectedErrorMessage);
+      whenIClickOnTheErrorSummaryLinkContainingText(...expectedErrorMessage);
+      thenMyFocusMovesTo("fixedVhfRadioInput");
+
+      finallyIClear("#fixedVhfRadioInput");
+      finallyIUncheck("#fixedVhfRadio");
+    });
```

Comment:
We are inconsistent in how we select DOM elements in `givenIHaveSelected`, `whenIType`, `finallyIClear` etc.  Some functions select based on the field `name`, others on the `id`.  

Cypress [best practices](https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements) guides using neither, instead we should use `data` tags.  I haven't done this in this PR to avoid yak shaving, but am considering it for some more work this afternoon!

Reply:
Interesting on the `data-` attributes.  They actually use them within the GDS library for identifying elements and running JS - it's how do the accordion and conditional displaying of checkboxes etc...

[MDN docs on this are also a good read](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes)

Like the idea though, try and plug it into the next set of tests to get a feel for how it might work across the existing test suite? I like this: 

![image](https://user-images.githubusercontent.com/76042279/110958927-19935480-8345-11eb-9551-70a89443aa4c.png)

The main reason I've seen against using `data-cy` attributes, is that they shouldn't be in your production build as they're only required for integration/e2e testing. There is [this](https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id) babel plugin that looks like it's specific for `data-test-id`, but does have an extension for defining custom attributes, so should be able to use this to strip `data-cy` attributes if we do want to remove them: 

![image](https://user-images.githubusercontent.com/76042279/110960059-43994680-8346-11eb-83a5-b33e6cf5c678.png)

Reply:
Interesting read!  I'm leaning towards _not_ keeping a parallel set of `data` tags because fields like `hexId` are unlikely to change as they refer to entities.  Adding a host of `data` tags and a node package to remove them for prod _feels_ like wasted effort.

![image](https://user-images.githubusercontent.com/54271433/110966737-7eeb4380-834d-11eb-8cbd-ad2c30ae0c57.png)

[This pattern](https://docs.cypress.io/api/commands/within.html#Examples) for selecting form elements looks good.

Will continue to feel it out.